code/modules/reagents/chemistry/holder/holder.dm

```diff
@@ -540,6 +542,9 @@
 			handle_reactions()
 		target_holder.handle_reactions()
 
+	if (target_atom)
+		SEND_SIGNAL(target_atom, COMSIG_ATOM_REAGENTS_TRANSFERRED_TO, src, transferred, total_transfered_amount, transferred_by, methods, ignore_stomach)
```

Comment:
This signal doesn't to exist. Since allergies can only be transferred when `method` var is set you can check the various `expose()` poc defined on `mob/living/carbon`. 

It has predefined signals you can hook onto

Reply:
Im not sure what you mean, carbon has a expose() proc? Or do you mean the expose_mob() proc on reagent? Or the COMSIG_REAGENT_EXPOSE_MOB?

Reply:
COMSIG_REAGENT_EXPOSE_MOB attaches to reagents, not atoms. Im not the best with understanding signals, but I opted to use this new signal because otherwise I think I'd have to attach a signal to the reagents of every cigarette in the game...? Correct me if I'm wrong, but this only listens for when it's host reagent is applied and not if an atom's host reagent gets transferred to.

Reply:
Check my comment in https://github.com/tgstation/tgstation/pull/92747#discussion_r2304678768 on how to use this